heat/core/signal.py

```diff
@@ -127,22 +149,54 @@ def convolve(a: DNDarray, v: DNDarray, mode: str = "full") -> DNDarray:
         signal = signal.to(float_type)
         weight = weight.to(float_type)
 
```

Comment:
Looks like `v` needs to be cast to float on GPU as well, GPU tests fail later at line 158
 
<img width="845" alt="Screenshot 2022-11-24 at 09 43 43" src="https://user-images.githubusercontent.com/39374113/203734658-420ff547-8e11-4b9a-bff2-a6af97fb5341.png">

Reply:
done, does it works now?